### ISSUE TITLE
CMSG_SPACE should be const

### ISSUE THREAD

**Issue by pacak** [rust-lang/libc#2819]
There was this: https://github.com/rust-lang/libc/issues/2087 but since https://github.com/rust-lang/libc/pull/2132 had no tests in it it either never worked or got broken at some point later:

```rust
use std::os::unix::prelude::RawFd;

const CONTROL_LEN: usize = unsafe { libc::CMSG_SPACE(std::mem::size_of::<RawFd>() as _) as _ };
```
```
t% cargo test
   Compiling libc v0.2.126
   Compiling asdf v0.1.0 (/home/pacak/rust/asdf)
error[E0015]: cannot call non-const fn `CMSG_SPACE` in constants
 --> src/lib.rs:3:37
  |
3 | const CONTROL_LEN: usize = unsafe { libc::CMSG_SPACE(std::mem::size_of::<RawFd>() as _) as _ };
  |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: calls in constants are limited to constant functions, tuple structs and tuple variants

For more information about this error, try `rustc --explain E0015`.
error: could not compile `asdff` due to previous error
warning: build failed, waiting for other jobs to finish...
error: build failed
```
rustc 1.60.0 (7737e0b5c 2022-04-04)

**Comment by JohnTitor** [comment]
You have to enable the `const-extern-fn` feature of the crate.

**Comment by pacak** [comment]
I see, and it's nightly only. That's sad.

Thank you.

**Comment by JohnTitor** [comment]
> I see, and it's nightly only. That's sad.

It'll be available from 1.62, thanks to https://github.com/rust-lang/rust/pull/95346 :)
